presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainFilterUtils.java

```diff
@@ -142,7 +153,7 @@ public static TupleDomainFilter toFilter(Domain domain)
      *
      * The logic is to return true if ranges are next to each other, but don't include the touch value.
      */
-    private static boolean isNanAllowed(List<Range> ranges)
+    private static boolean isNotIn(List<Range> ranges)
```

Comment:
method comment needs to be updated and discussion of NaNs needs to be moved to the caller; I'd move changes to this method into a separate commit 

```
    /**
     * Returns true is ranges represent != or NOT IN filter for double, float or string column.
     *
     * The logic is to return true if ranges are next to each other, but don't include the touch value.
     */
    private static boolean isNotIn(List<Range> ranges)
```

and, in the caller:

```

        if (firstRangeFilter instanceof DoubleRange || firstRangeFilter instanceof FloatRange) {
            // != and NOT IN filters should return true when applied to NaN
            // E.g. NaN != 1.0 as well as NaN NOT IN (1.0, 2.5, 3.6) should return true; otherwise false.
            boolean nanAllowed = isNotIn(ranges);
            return MultiRange.of(rangeFilters, nullAllowed, nanAllowed);
        }

        return MultiRange.of(rangeFilters, nullAllowed, false);
```